worldedit-core/src/main/java/com/fastasyncworldedit/core/extent/processor/heightmap/HeightmapProcessor.java

```diff
@@ -86,8 +86,9 @@ public IChunkSet processSet(IChunk chunk, IChunkGet get, IChunkSet set) {
                         } else if (getSection == null) {
                             getSection = get.load(layer);
                             // skip empty layer
-                            if (Arrays.equals(getSection, FaweCache.INSTANCE.EMPTY_CHAR_4096)
-                                    || Arrays.equals(getSection, AIR_LAYER)) {
+                            if (getSection == null ||
```

Comment:
`Arrays.equals` already covers that in this case, although it might make sense to have this check, as other overloads of that method would indeed throw a NPE. That said, it would probably make sense to have a comment about it in the code.

Reply:
That's true. The NPE in the issue was probably a race condition